[BE] remove torch deploy - conditionals

PaliC opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #158407
* #158291
* #158290
* __->__ #158288

This PR is part of the work to deprecate torch::deploy in OSS. Effectively it does 3 things to get started.
1. Remove test_deploy_interaction as we no longer need to worry about this
2. Remove all torch._running_with_deploy checks and use the False path always (surfaced 1)
3. Remove `USE_DEPLOY` and switch to the default path always

Note: MyPy does fail on a bunch of things here as a bunch of older files are touched. It may be better to fix these things on a separate PR


cc @H-Huang @awgu @wanchaol @fegin @fduwjj @wz337 @wconstab @d4l3k @pragupta @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @chenyang78 @kadeng @muchulee8 @amjames @chauhang @aakhundov @coconutruben @Lucaskabela

PaliC committed:
Update

[ghstack-poisoned]

PaliC committed:
Update

[ghstack-poisoned]

PaliC committed:
Update

[ghstack-poisoned]

PaliC committed:
Update

[ghstack-poisoned]

albanD commented on torch/csrc/utils/python_dispatch.cpp:
```diff
@@ -193,9 +193,6 @@ class PythonKernelHolder : public c10::OperatorKernel {
     // Eventually just delete this entirely.  (Note that you may break
     // multipy anyway this way with dispatcher  // codespell:ignore multipy
     // registered functions that require hermetic to be off.)
-#if defined(USE_DEPLOY)
```
The comment above can go !

albanD commented on torch/distributed/_functional_collectives.py:
```diff
@@ -19,23 +19,16 @@
 
 
-if torch._running_with_deploy():
+try:
+    from torch.compiler import is_dynamo_compiling as is_torchdynamo_compiling
+except Exception:
+    warnings.warn(
+        "Unable to import torchdynamo util `is_torchdynamo_compiling`, so won't support torchdynamo correctly"
+    )
```
@jansel this looks pretty bad right?

albanD reviewed (approved):
Very nice!

PaliC committed:
Update

[ghstack-poisoned]

PaliC committed:
Update on "[BE] remove torch deploy - conditionals"


This PR is part of the work to deprecate torch::deploy in OSS. Effectively it does 3 things to get started.
1. Remove test_deploy_interaction as we no longer need to worry about this
2. Remove all torch._running_with_deploy checks and use the False path always (surfaced 1)
3. Remove `USE_DEPLOY` and switch to the default path always

Note: MyPy does fail on a bunch of things here as a bunch of older files are touched. It may be better to fix these things on a separate PR


cc H-Huang awgu wanchaol fegin fduwjj wz337 wconstab d4l3k voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov 

[ghstack-poisoned]

PaliC committed:
Update

[ghstack-poisoned]

XuehaiPan commented on torch/__init__.py:
```diff
@@ -60,14 +50,9 @@ def _running_with_deploy() -> builtins.bool:
     USE_GLOBAL_DEPS,
     USE_RTLD_GLOBAL_WITH_LIBTORCH,
 )
+from torch.torch_version import __version__ as __version__
```
```suggestion
from torch import version as version
from torch.torch_version import __version__ as __version__
```

PaliC replied:
can you do this in a separate PR, I am not too sure how safe this is, and I don't want to keep running ci on this stack as it seems to be pretty good otherwise

XuehaiPan commented on torch/__init__.py:
```diff
@@ -34,20 +34,10 @@
-from . import version
-
 
 if TYPE_CHECKING:
     from .types import Device, IntLikeType
```
This can be moved after the import statements now.

PaliC replied:
what do you mean here?

PaliC committed:
Update on "[BE] remove torch deploy - conditionals"


This PR is part of the work to deprecate torch::deploy in OSS. Effectively it does 3 things to get started.
1. Remove test_deploy_interaction as we no longer need to worry about this
2. Remove all torch._running_with_deploy checks and use the False path always (surfaced 1)
3. Remove `USE_DEPLOY` and switch to the default path always

Note: MyPy does fail on a bunch of things here as a bunch of older files are touched. It may be better to fix these things on a separate PR


cc H-Huang awgu wanchaol fegin fduwjj wz337 wconstab d4l3k voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov 

[ghstack-poisoned]

PaliC committed:
Update

[ghstack-poisoned]

PaliC committed:
Update

[ghstack-poisoned]

PaliC committed:
Update on "[BE] remove torch deploy - conditionals"


This PR is part of the work to deprecate torch::deploy in OSS. Effectively it does 3 things to get started.
1. Remove test_deploy_interaction as we no longer need to worry about this
2. Remove all torch._running_with_deploy checks and use the False path always (surfaced 1)
3. Remove `USE_DEPLOY` and switch to the default path always

Note: MyPy does fail on a bunch of things here as a bunch of older files are touched. It may be better to fix these things on a separate PR


cc H-Huang awgu wanchaol fegin fduwjj wz337 wconstab d4l3k voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov 

[ghstack-poisoned]

ZainRizvi commented:
@pytorchbot revert -c ghfirst -m "Sorry but this is breaking internally (see D78496147). To validate your fixes internally, you can follow the instructions here: https://fburl.com/fixing-ghfirst-reverts"

PaliC committed:
Update

[ghstack-poisoned]

PaliC commented:
@pytorchbot revert

ZainRizvi commented:
@pytorchbot revert -c ghfirst -m "Reverting as per offline discussion to fix internal breaks.  @PaliC will reland this as a codev diff. Instructions here: https://fburl.com/fixing-ghfirst-reverts"

PaliC committed:
Update on "[BE] remove torch deploy - conditionals"


This PR is part of the work to deprecate torch::deploy in OSS. Effectively it does 3 things to get started.
1. Remove test_deploy_interaction as we no longer need to worry about this
2. Remove all torch._running_with_deploy checks and use the False path always (surfaced 1)
3. Remove `USE_DEPLOY` and switch to the default path always

Note: MyPy does fail on a bunch of things here as a bunch of older files are touched. It may be better to fix these things on a separate PR


cc H-Huang awgu wanchaol fegin fduwjj wz337 wconstab d4l3k pragupta voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov coconutruben Lucaskabela 

[ghstack-poisoned]

PaliC commented:
@pytorchbot merge
